addons/mail/models/mail_channel.py

```diff
@@ -1317,24 +1317,26 @@ def channel_create(self, name, privacy='public'):
         return channel_info
 
     @api.model
-    def group_chat_create(self, partners_to):
+    def group_chat_create(self, current_partner, partners_to):
         """ Create a private group chat channel.
+            :param current_partner : res.partner id who created the conversation
             :param partners_to : list of res.partner ids to add to the conversation
             :returns: channel_info of the created or existing channel
             :rtype: dict
         """
-        if self.env.user.partner_id.id not in partners_to:
-            partners_to.append(self.env.user.partner_id.id)
+        if current_partner not in partners_to:
+            partners_to.append(current_partner)
         channel = self.create({
             'channel_partner_ids': [Command.link(partner_id) for partner_id in partners_to],
             'public': 'private',
             'channel_type': 'group',
             'email_send': False,
             'name': ', '.join(self.env['res.partner'].sudo().browse(partners_to).mapped('name')),
         })
-        self.env['mail.channel.partner'].search(
-            [('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', channel.id)]).write(
-            {'is_pinned': True})
+        for partner_id in partners_to:
+            self.env['mail.channel.partner'].search(
+                [('partner_id', '=', partner_id), ('channel_id', '=', channel.id)]).write(
+                {'is_pinned': True})
```

Comment:
This is equivalent (and won't do extra queries): `channel.channel_last_seen_partner_ids.write({'is_pinned': True})`